File: app/src/main/java/com/example/junedshaikh_project/fragment/MoreFragment.java
package com.example.junedshaikh_project.fragment;

import android.content.Intent;
import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import com.example.junedshaikh_project.activity.LoginActivity;
import com.example.junedshaikh_project.databinding.FragmentMoreBinding;
import com.google.firebase.auth.FirebaseAuth;
import com.google.firebase.auth.FirebaseUser;

public class MoreFragment extends Fragment {

    private FragmentMoreBinding binding;
    private FirebaseAuth auth;

    @Nullable
    @Override
    public View onCreateView(@NonNull LayoutInflater inflater, @Nullable ViewGroup container, @Nullable Bundle savedInstanceState) {
        binding = FragmentMoreBinding.inflate(inflater, container, false);
        View view = binding.getRoot();
        return view;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        auth = FirebaseAuth.getInstance();

        FirebaseUser currentUser = auth.getCurrentUser();
        if (currentUser != null) {
            binding.nameTextView.setText(currentUser.getDisplayName());
            binding.emailTextView.setText(currentUser.getEmail());
            binding.passwordTextView.setText("********");
        }

        binding.logoutButton.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                auth.signOut();
                startActivity(new Intent(getActivity(), LoginActivity.class));
                if (getActivity() != null) {
                    getActivity().finish();
                }
            }
        });
    }
}

File: app/src/main/java/com/example/junedshaikh_project/fragment/MoreFragment.kt
package com.example.junedshaikh_project.fragment

import android.content.Intent
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.core.content.ContextCompat.startActivity
import com.example.junedshaikh_project.activity.LoginActivity
import com.example.junedshaikh_project.databinding.FragmentMoreBinding
import com.example.thefoodcoast.base.BaseFragment
import com.google.firebase.auth.FirebaseAuth

class MoreFragment : BaseFragment<FragmentMoreBinding>() {

    private lateinit var auth: FirebaseAuth

    override fun getFragmentBinding(
        inflater: LayoutInflater,
        container: ViewGroup?
    ) = FragmentMoreBinding.inflate(inflater, container, false)


    override fun initialSetUp() {
        super.initialSetUp()
        auth = FirebaseAuth.getInstance()

        var currentUser = auth.currentUser
        binding.nameTextView.text = currentUser?.displayName
        binding.emailTextView.text = currentUser?.email
        binding.passwordTextView.text = "********"

        binding.logoutButton.setOnClickListener {
            auth.signOut()
            startActivity(Intent(activity, LoginActivity::class.java))
            activity?.finish()
        }
    }
}